src/services.ts

```diff
@@ -0,0 +1,121 @@
+import { createAuthHelpers } from "./redux/authHelpers";
+import * as Rx from "rxjs";
+import * as RxOp from "rxjs/operators";
+import { ajax, AjaxResponse } from "rxjs/ajax";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+import { PathReporter } from "io-ts/es6/PathReporter";
+import { pipe } from "fp-ts/lib/function";
+
+export const createService = ({
+  baseUrl,
+  authHelpers,
+  scopes,
+}: {
+  baseUrl: string;
+  authHelpers: ReturnType<typeof createAuthHelpers>;
+  scopes: string[];
+}) => {
+  let languageSelector = () => "";
+  const makeRequest = <A>(
+    decoder: t.Type<A>,
+    fn: (token: string) => Rx.Observable<AjaxResponse>
+  ) =>
+    pipe(
+      authHelpers.getTokenSilently(scopes, authHelpers.getUserAccount()!),
+      TE.chain(
+        (token) => () =>
+          pipe(
+            fn(token.accessToken)
+              .pipe(
+                RxOp.map((x) => x.response),
+                RxOp.map(decoder.decode),
+                RxOp.tap(
+                  E.fold(
+                    (e) => {
+                      console.log(PathReporter.report(t.failures(e)));
+                    },
+                    () => {}
+                  )
+                ),
+                RxOp.map(E.mapLeft(() => "parser error")),
+                RxOp.catchError((e) => Rx.of(E.left<string, A>(e)))
+              )
+              .toPromise()
+          )
+      )
+    );
+  return {
+    updateLanguage: (ipdatedlanguageSelector: () => string) => {
+      languageSelector = ipdatedlanguageSelector;
+    },
+    get: <A>(url: string, decoder: t.Type<A>, headers?: t.Any) =>
+      makeRequest(decoder, (token) =>
+        ajax.get(`${baseUrl}/v1.0/${url}`, {
```

Comment:
why v1.0 is hardcoded in a generic 'service' module? shouldn't be either part of baseurl or passed in by caller?